contracts/PBAB+Collabs/GenArt721CoreV2WithEvents.sol

```diff
@@ -0,0 +1,758 @@
+    }
+
+    /**
+     * @notice Updates price per token of project `_projectId` to be
+     * '_pricePerTokenInWei`, in Wei.
+     */
+    function updateProjectPricePerTokenInWei(
+        uint256 _projectId,
+        uint256 _pricePerTokenInWei
+    ) public onlyArtist(_projectId) {
+        projectIdToPricePerTokenInWei[_projectId] = _pricePerTokenInWei;
+
+        emit ProjectUpdated(_projectId, "projectIdToPricePerTokenInWei");
+    }
+
+    /**
+     * @notice Updates name of project `_projectId` to be `_projectName`.
+     */
+    function updateProjectName(uint256 _projectId, string memory _projectName)
+        public
+        onlyUnlocked(_projectId)
+        onlyArtistOrWhitelisted(_projectId)
+    {
+        projects[_projectId].name = _projectName;
+
+        emit ProjectUpdated(_projectId, "projectName");
+    }
+
+    /**
+     * @notice Updates artist name for project `_projectId` to be
+     * `_projectArtistName`.
+     */
+    function updateProjectArtistName(
+        uint256 _projectId,
+        string memory _projectArtistName
+    ) public onlyUnlocked(_projectId) onlyArtistOrWhitelisted(_projectId) {
+        projects[_projectId].artist = _projectArtistName;
+        emit ProjectUpdated(_projectId, "projectName");
+    }
+
+    /**
+     * @notice Updates additional payee for project `_projectId` to be
+     * `_additionalPayee`, receiving `_additionalPayeePercentage` percent
+     * of artist mint and royalty revenues.
+     */
+    function updateProjectAdditionalPayeeInfo(
+        uint256 _projectId,
+        address payable _additionalPayee,
+        uint256 _additionalPayeePercentage
+    ) public onlyArtist(_projectId) {
+        require(_additionalPayeePercentage <= 100, "Max of 100%");
+        projectIdToAdditionalPayee[_projectId] = _additionalPayee;
+        projectIdToAdditionalPayeePercentage[
+            _projectId
+        ] = _additionalPayeePercentage;
+
+        emit ProjectUpdated(_projectId, "additionalPayeeInfo");
+    }
+
+    /**
+     * @notice Updates artist secondary market royalties for project
+     * `_projectId` to be `_secondMarketRoyalty` percent.
+     */
+    function updateProjectSecondaryMarketRoyaltyPercentage(
+        uint256 _projectId,
+        uint256 _secondMarketRoyalty
+    ) public onlyArtist(_projectId) {
+        require(_secondMarketRoyalty <= 100, "Max of 100%");
+        projectIdToSecondaryMarketRoyaltyPercentage[
+            _projectId
+        ] = _secondMarketRoyalty;
+
+        emit ProjectUpdated(_projectId, "secondaryMarketRoyaltyPercentage");
+    }
+
+    /**
+     * @notice Updates description of project `_projectId`.
+     */
+    function updateProjectDescription(
+        uint256 _projectId,
+        string memory _projectDescription
+    ) public onlyArtist(_projectId) {
+        projects[_projectId].description = _projectDescription;
+
+        emit ProjectUpdated(_projectId, "projectDescription");
```

Comment:
In general I'd remove the project "project" prefix from the update field name since it's redundant considering the event name.